elastica/dissipation.py

```diff
@@ -220,19 +223,85 @@ def __init__(self, filter_order: int, **kwargs):
                 "Invalid filter order! Filter order must be a positive integer."
             )
         self.filter_order = filter_order
-        self.velocity_filter_term = np.zeros_like(self._system.velocity_collection)
-        self.omega_filter_term = np.zeros_like(self._system.omega_collection)
+
+        if self._system.ring_rod_flag:
+            # There are two periodic boundaries
+            blocksize = self._system.n_elems + 2
+            self.velocity_filter_term = np.zeros((3, blocksize))
+            self.omega_filter_term = np.zeros((3, blocksize))
+
+            @njit(cache=True)
+            def filter_function_generator(
```

Comment:
Maybe pull this part of the code outside the class? We can name them `_filter_function_generator_periodic_condition`, and set `self.filter_function = _filter_function_generator_periodic_condition`